### DISCUSSION TITLE
How to integrate Github login with firebase auth in both the web and mobile?

### DISCUSSION THREAD

**Discussion by KittichoteKamalapirat** [expo/expo#29000] [Archived: Questions]
### Issue
I created a project in Firebase which consists of 3 apps, the `web` (NextJS), `ios` app (React Native Expo), and `android` (React Native Expo). I added Github as a provider in Firebase Auth and also added a new OAuth app in Github. However, when adding a new OAuth app in Github, I have to provide an **Authorization Callback URL** which is the problem here. For the web, I have to fill in `https://myapp.firebaseapp.com/__/auth/handler` but for the mobile app, I have to fill in `myapp://` (`myapp` is the scheme field in expo's app.json) which is a deeplink to my mobile app. **However, I can only provide 1 redirect url**. How should I get away with this limitation?

**Basically,**
1. The login on mobile works perfectly when I provide `myapp://` as the authorization callback URL on Github. But the web would break.
2. Simiarly, the login on web also works perfectly when I provide `https://myapp.firebaseapp.com/__/auth/handler` as the authorization callback URL on Github. But the mobile app would break.
3. **The problem is** I can't get both of them to work at the same time.

#### Github OAuth Setting

[![Github OAuth Setting][1]][1]

#### Firebase's Github Provider Config

[![Firebase's Github Provider Config][3]][3]

  [1]: https://i.sstatic.net/nubh02oP.png
  [2]: https://i.sstatic.net/DdK8WXU4.png
  [3]: https://i.sstatic.net/lQ0RT2u9.png

**Comment by LizMyers** [comment]
Create separate apps per platform.
